src/screens/WasteCollectionScreen.js

```diff
@@ -143,6 +132,79 @@ export const WasteCollectionScreen = ({ navigation }) => {
     }
   );
 
+  const wasteAddressesStreetCount = globalSettings?.settings?.wasteAddresses?.streetCount || 5;
+  const wasteAddressesTwoStep = !!globalSettings?.settings?.wasteAddresses?.twoStep;
+
+  const getStreetString = useCallback(
+    (address) => {
+      if (wasteAddressesTwoStep) {
+        return address.street || '';
+      }
+
+      return `${address.street} (${address.zip} ${address.city})`;
+    },
+    [wasteAddressesTwoStep]
+  );
+
+  // show cities that contain the string in it
+  // return empty list on an exact match (except for capitalization)
+  const filterCities = useCallback(
+    (currentInputValue, addressesData) => {
+      if (currentInputValue === '') return [];
+
+      const cities = addressesData?.filter(
+        (address) =>
+          address.city !== currentInputValue &&
+          address.city.toLowerCase().includes(currentInputValue.toLowerCase())
+      );
+
+      return _sortBy(_uniqBy(cities, 'city'), 'city');
+    },
+    [getStreetString]
+  );
+
+  // show streets that contain the string in it
+  // consider the city if it is set because of the two step process
+  // return empty list on an exact match (except for capitalization)
+  const filterStreets = useCallback(
+    (currentInputValue, addressesData) => {
+      if (wasteAddressesTwoStep && inputValueCity === '') return [];
+      if (currentInputValue === '') return [];
+
```

Comment:
no, brackets would be missing in your example around `wasteAddressesTwoStep && inputValueCity === ''`.
and i think it is not a better solution. i see different scenarios here. having two step, than it is necessary to check the `inputValueCity`. in every other case, like not having two step or two step with `inputValueCity !== ''`, `currentInputValue === ''` needs to be checked. because of this two point of view, i decided to have two conditions with return here.

Reply:
then we do it this way. 🎉